samples/graph/vehicle_core_provider/src/main.rs

```diff
@@ -0,0 +1,320 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+// SPDX-License-Identifier: MIT
+
+mod request_impl;
+
+use digital_twin_model::sdv_v1 as sdv;
+use env_logger::{Builder, Target};
+use log::{info, LevelFilter};
+use parking_lot::Mutex;
+use samples_common::constants::{digital_twin_operation, digital_twin_protocol};
+use samples_common::provider_config;
+use samples_common::utils::retrieve_invehicle_digital_twin_uri;
+use samples_protobuf_data_access::async_rpc::v1::request::request_server::RequestServer;
+use samples_protobuf_data_access::digital_twin_registry::v1::digital_twin_registry::digital_twin_registry_client::DigitalTwinRegistryClient;
+use samples_protobuf_data_access::digital_twin_registry::v1::digital_twin_registry::{
+    EndpointInfo, EntityAccessInfo, RegisterRequest, RegisterResponse,
+};
+use std::collections::HashMap;
+use std::net::SocketAddr;
+use std::sync::Arc;
+use tonic::{transport::Server, Status};
+use tokio_retry::Retry;
+use tokio_retry::strategy::{ExponentialBackoff, jitter};
+
+use crate::request_impl::{InstanceData, ProviderState, RequestImpl};
+
+/// The base duration in milliseconds for the exponential backoff strategy.
+const BACKOFF_BASE_DURATION_IN_MILLIS: u64 = 100;
+
+/// The maximum number of retries.
+const MAX_RETRIES: usize = 100;
+
+/// Add an entry to the instance map.
+/// # Arguments
+/// * `instance_map` - The instance map.
+/// * `instance_id` - The instance id.
+/// * `model_id` - The model id.
+/// * `description` - The description.
+/// * `serialized_value` - The serialized value.
+fn add_entry_to_instance_map(
+    instance_map: &mut HashMap<String, InstanceData>,
+    instance_id: String,
+    model_id: String,
+    description: String,
+    serialized_value: String,
+) {
+    instance_map.insert(instance_id, InstanceData { model_id, description, serialized_value });
+}
+
+/// Create the provider's state.
+fn create_provider_state() -> ProviderState {
+    let mut result: ProviderState = ProviderState { instance_map: HashMap::new() };
+
+    // Create the seat massager ids.
+
+    let front_left_airbag_seat_massager_instance_id = "front_left_airbag_seat_massager".to_string();
+
+    let front_right_airbag_seat_massager_instance_id =
+        "front_right_airbag_seat_massager".to_string();
+
+    let back_left_airbag_seat_massager_instance_id = "back_left_airbag_seat_massager".to_string();
+
+    let back_center_airbag_seat_massager_instance_id =
+        "back_center_airbag_seat_massager".to_string();
+
+    let back_right_airbag_seat_massager_instance_id = "back_right_airbag_seat_massager".to_string();
```

Comment:
should these be shared constants since they're used by both the provider and the consumer?

Reply:
Actually, they should be in a config file.  I'll try to improve on this.

Reply:
I can make them constants, but I don't want to share them as constants, as they really should be in a shared config file.

Reply:
Here is the requested issue: https://github.com/eclipse-ibeji/ibeji/issues/120

Reply:
Addressed.